hi-ml-cpath/testhisto/testhisto/models/test_transforms.py

```diff
@@ -89,6 +90,31 @@ def test_load_tiles_batch() -> None:
     assert_dicts_equal(bagged_loaded_batch, loaded_bagged_batch)
 
 
+@pytest.mark.parametrize("scale_intensity", [True, False])
+def test_itensity_scaling_load_tiles_batch(scale_intensity: bool, mock_panda_tiles_root_dir: Path) -> None:
+    tiles_dataset = PandaTilesDataset(mock_panda_tiles_root_dir)
+    image_key = tiles_dataset.IMAGE_COLUMN
+    max_bag_size = 4
+    bagged_dataset = BagDataset(tiles_dataset, bag_ids=tiles_dataset.slide_ids,  # type: ignore
+                                max_bag_size=max_bag_size)
+    load_batch_transform = LoadTilesBatchd(image_key, scale_intensity=scale_intensity)
+    index = 0
+
+    # Test that the transform returns images in [0, 255] range
+    bagged_batch = bagged_dataset[index]
+    manually_loaded_batch = load_batch_transform(bagged_batch)
+
+    pixels_dtype = torch.uint8 if not scale_intensity else torch.float32
+    max_val = 255 if not scale_intensity else 1.
+
+    bag_size = manually_loaded_batch[image_key][index].shape[0] - 1
+    index = min(np.random.randint(0, max_bag_size), bag_size)
```

Comment:
I would just work with a fixed index (either check all entries in the bag, or just a single one - but if single, don't randomize. Randomization adds obvious problems with reproducibility of test failures.)